zookeeper/source/zk_dn_client.cc

```diff
@@ -84,64 +84,81 @@ namespace zkclient{
 
 		LOG(INFO) << "About to create the znode for the data ip";
 		if (zk->exists(BLOCK_LOCATIONS + std::to_string(uuid), exists, error_code)) {
-        	if (exists) {
-        		// TODO: Add to spec the location to which we're writing the size
-        		// zk->set(BLOCK_LOCATIONS + std::to_string(uuid), ZKWrapper::EMPTY_VECTOR, error_code); 
+			if (exists) {
+				// TODO: Add to spec the location to which we're writing the size
+				// zk->set(BLOCK_LOCATIONS + std::to_string(uuid), ZKWrapper::EMPTY_VECTOR, error_code);
 				// Add this datanode as the block's location in block_locations
 				DataNodeZNode znode_data;
 				strcpy(znode_data.ipPort, id.c_str());
 				std::vector<std::uint8_t> data(sizeof(znode_data));
 				memcpy(&data[0], &znode_data, sizeof(znode_data));
 				LOG(INFO) << BLOCK_LOCATIONS + std::to_string(uuid) + "/" + id;
-				if(!zk->create(BLOCK_LOCATIONS + std::to_string(uuid) + "/" + id, ZKWrapper::EMPTY_VECTOR, error_code, false)) {
+				if(!zk->create(BLOCK_LOCATIONS + std::to_string(uuid) + "/" + id, data, error_code, false)) {
 					LOG(ERROR) << CLASS_NAME <<  "Failed creating /block_locations/<block_uuid>/<block_id> " << error_code;
 					return false;
 				}
-        	}
+			}
 			else {
 				LOG(ERROR) << CLASS_NAME << "/block_locations/<block_uuid> did not exist " << error_code;
 				return false;
 			}
 		}
 
+
+		// Write the block size to health
+		if (zk->exists(HEALTH_BACKSLASH + id + BLOCKS, exists, error_code)) {
```

Comment:
Sorry, I forgot to mention this to you @agardella -- I realized that this is not a good location to put the block size because we would be recording the block size REPLICATION_FACTOR number of times. It's better to put it as the payload for BLOCK_LOCATIONS + uuid, like in line 89.